### PR TITLE
Type conversation

### PR DESCRIPTION
**Pull request by raditya3**
### Description

Added type check in following files

- assets/src/components/conversations/ConversationDetailsSidebar.tsx
- assets/src/components/conversations/ConversationHeader.tsx
- assets/src/components/conversations/ConversationItem.tsx
- assets/src/components/conversations/ConversationMessages.tsx

Modified types in
- assets/src/types.ts:
     Changed Users, Customer, Message, Conversation type

### Issue

#308 

### Screenshots

image

(Note : Console error due to my ad-blocker)

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: assets/src/types.ts [hunk 1]
```diff
@@ -2,17 +2,32 @@ export type User = {
   id: number;
   email: string;
   display_name?: string;
-  full_name?: string;
+  name?: string;
```
Comment (reichert621) [inline assets/src/types.ts:5]: why this change? I'm not sure this is correct 🤔

**Review by reichert621 — CHANGES_REQUESTED** [review]
I'm not sure the change from `full_name` -> `name` is correct, but let me know if I'm missing something!

File: assets/src/types.ts (hunk 1 above)
Comment (raditya3) [reply assets/src/types.ts:5]: image

The BE API returns response having Customer object having property as `name` (not `full_name`), so to maintain uniformity (and to replace `any` with `Customer`), I replaced the variable name

Comment (reichert621) [reply assets/src/types.ts:5]: ah -- `name` is for the `Customer` type, but `full_name` is for the `User` type

Comment (raditya3) [reply assets/src/types.ts:5]: https://github.com/papercups-io/papercups/blob/4e9c4580faa7641ccc3adb338456cbcfeca0ab17/assets/src/components/conversations/ConversationHeader.tsx#L198
In this line `user.name` was used, that's why I must have changed the property. I think I should keep the property as `full_name` and change the corresponding line

Comment (reichert621) [reply assets/src/types.ts:5]: ah yes, that line is incorrect 👍

Comment (raditya3) [reply assets/src/types.ts]: Can you review again?

**Review by reichert621 — APPROVED** [review]
thanks @raditya3!
